src/java/org/apache/cassandra/utils/Overlaps.java

```diff
@@ -123,44 +125,138 @@ public static <E, B> List<B> assignOverlapsIntoBuckets(int threshold,
                                                            List<Set<E>> overlaps,
```

Comment:
We could add something explaining the prioritization to the JavaDoc, e.g. "For non-transitive inclusionMethod the order in which we select the buckets matters because an sstables that spans overlap sets could be chosen for only one of the candidate buckets containing it. To make the most efficient selection we thus perform it by descending size, starting with the sets with most overlap."

If the MarkDown doc explains this in detail, we should change that too (I don't think it does).